src/models/outbound/pispOTPValidate.interface.ts

```diff
@@ -21,22 +21,46 @@
  * Gates Foundation
  - Name Surname <name.surname@gatesfoundation.com>
 
- - Kevin Leyow <kevin.leyow@modusbox.com>
+ - Kevin Leyow - kevin.leyow@modusbox.com
  --------------
  ******/
+import {
+  ControlledStateMachine,
+  PersistentModelConfig, StateData
+} from '~/models/persistent.model'
+import { Method } from 'javascript-state-machine'
+import { ThirdpartyRequests } from '@mojaloop/sdk-standard-components';
+import {
+  thirdparty as tpAPI
+} from '@mojaloop/api-snippets'
+import { PubSub } from '~/shared/pub-sub'
 
-// TODO: use Async2sync model for this.
-/**
- * @name channelName
- * @description generates the pub/sub channel name
- * @param {object} - args
- * @param {string} args.consentRequestId - the consent request  id
- * @returns {string} - the pub/sub channel name
- */
-export function notificationChannel (consentRequestId: string) {
-  if (!consentRequestId) {
-    throw new Error('PISPConsentRequest.notificationChannel: \'consentRequestId\' parameter is required')
-  }
-  // channel name
-  return `consent_request_${consentRequestId}`
+export enum PISPOTPValidateModelState {
```

Comment:
we can use here a type `OutboundAPI.Schemas.ConsentRequestsValidateState` so the enumeration values will come from API definition and there will be no need to redefine them here

Reply:
How do I turn an interface into a type? or enum in this case?

Reply:
- simply replace `interface` with `type` keyword
- to replace `enum` by type is little more complex -> you can use an alternate type operator ->  `type X = 'A' | '1'` -> but then you cannot use dot notation anymore -> you are using the values.

Reply:
to use `ConsentRequestsValidateState` you need to reimport it from generated openapi.d.ts file into `interface/outbound/api-interface/index.ts` example https://github.com/mojaloop/thirdparty-scheme-adapter/blob/master/src/interface/outbound/api_interfaces/index.ts#L57 so I can use it here: https://github.com/mojaloop/thirdparty-scheme-adapter/blob/master/src/models/pispTransaction.interface.ts#L50

Reply:
Maybe not complex but different ;-) For `enum addicted` persons using strict values instead enumType.dot notation would be strange, but TS doesn't allow you to assign invalid strict values what makes dot notation obsolete and too elaborative. And you can write much less code without dot notation.
Yes, you are right `PISPTransactionModelState` is the leftover from my last refactoring where I migrated from enums to autogenerated types & strict values - to be honest, I am considering making `StateData` from `persistent.model` a generic one, parametrized by state type so then we will have full type checking ;-)

Reply:
changed to raw strings using `currentState: 'start' | 'OTPIsValid' | 'errored',` instead. done.